lib/xcodeproj/scheme/command_line_arguments.rb

```diff
@@ -0,0 +1,154 @@
+
+module Xcodeproj
+  class XCScheme
+    COMMAND_LINE_ARGS_NODE = 'CommandLineArguments'.freeze
+    COMMAND_LINE_ARG_NODE = 'CommandLineArgument'.freeze
+
+    # This class wraps the CommandLineArguments node of a .xcscheme XML file. This
+    # is just a container of CommandLineArgument objects. It can either appear on a
+    # LaunchAction or TestAction scheme group.
+    #
+    class CommandLineArguments < XMLElementWrapper
+      # @param [nil,REXML::Element,Array<CommandLineArgument>,Array<Hash{Symbol => String,Bool}>] node_or_arguments
+      #        The 'CommandLineArguments' XML node, or list of command line arguments, that this object represents.
+      #          - If nil, an empty 'CommandLineArguments' XML node will be created
+      #          - If an REXML::Element, it must be named 'CommandLineArguments'
+      #          - If an Array of objects or Hashes, they'll each be passed to {#assign_argument}
+      #
+      def initialize(node_or_arguments = nil)
+        create_xml_element_with_fallback(node_or_arguments, COMMAND_LINE_ARGS_NODE) do
+          @all_arguments = []
+          node_or_arguments.each { |var| assign_argument(var) } unless node_or_arguments.nil?
+        end
+      end
+
+      # @return [Array<CommandLineArgument>]
+      #         The key value pairs currently set in @xml_element
+      #
+      def all_arguments
+        @all_arguments ||= @xml_element.get_elements(COMMAND_LINE_ARG_NODE).map { |argument| CommandLineArgument.new(argument) }
+      end
+
+      # Adds a given argument to the set of command line arguments, or replaces it if that key already exists
+      #
+      # @param [CommandLineArgument,Hash{Symbol => String,Bool}] argument
+      #        The argument to add or update, backed by an CommandLineArgument node.
+      #          - If an CommandLineArgument, the previous reference will still be valid
+      #          - If a Hash, must conform to {CommandLineArgument#initialize} requirements
+      # @return [Array<CommandLineArgument>]
+      #         The new set of command line arguments after addition
+      #
+      def assign_argument(argument)
+        env_var = argument.is_a?(CommandLineArgument) ? argument : CommandLineArgument.new(argument)
+        all_arguments.each { |existing_var| remove_argument(existing_var) if existing_var.argument == env_var.argument }
+        @xml_element.add_element(env_var.xml_element)
+        @all_arguments << env_var
+      end
+
+      # Removes a specified argument (by string or object) from the set of command line arguments
+      #
+      # @param [CommandLineArgument,String] argument
+      #        The argument to remove
+      # @return [Array<CommandLineArgument>]
+      #         The new set of command line arguments after removal
+      #
+      def remove_argument(argument)
+        env_var = argument.is_a?(CommandLineArgument) ? argument : all_arguments.find { |var| var.argument == argument }
```

Comment:
Updated as well. I have some ideas on how to clean this (and `EnvironmentVariables` up a bit so less of this is needed, but I'll send another pull request since that's a larger set of changes).